src/wallet/rpcwallet.cpp

```diff
@@ -162,11 +152,6 @@ UniValue getnewaddress(const UniValue& params, bool fHelp)
 
     LOCK2(cs_main, pwalletMain->cs_wallet);
 
-    // Parse the account first so we don't generate a key if there's an error
-    string strAccount;
-    if (params.size() > 0)
-        strAccount = AccountFromValue(params[0]);
```

Comment:
Removing this check while leaving `params.size() > 1` above means that this parameter position can now be set arbitrarily. We should continue to enforce that this is the empty string if present.